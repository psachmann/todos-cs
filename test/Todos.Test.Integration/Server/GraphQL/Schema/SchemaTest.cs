namespace Todos.Server.GraphQL;

public class SchemaTest
{
    [Fact]
    public async Task SchemeChangeTest()
    {
        var schema = await new ServiceCollection()
            .AddGraphQLServer()
            .AddQueryType<QueryType>()
            .BuildSchemaAsync();

        schema.Should().NotBeNull();
        schema.ToString().MatchSnapshot();
    }
}
