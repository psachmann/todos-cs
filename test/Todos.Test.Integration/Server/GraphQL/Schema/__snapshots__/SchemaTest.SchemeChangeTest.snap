schema {
  query: Query
}

type Query {
  todoItem: TodoItem
  todoItems: [TodoItem]
}

"The representation of a todo item."
type TodoItem {
  "The id of the todo item."
  id: UUID!
  "The creation date of the todo item."
  createdAt: DateTime!
  "The last updated date of the todo item."
  updatedAt: DateTime!
  "The title of the todo item.\nCan be an empty string."
  title: String!
  "The description of the todo item.\nCan be an empty string."
  description: String!
  "Indicates whether the todo item is done or not.\nDefault value is false."
  isDone: Boolean!
}

"The `@specifiedBy` directive is used within the type system definition language to provide a URL for specifying the behavior of custom scalar definitions."
directive @specifiedBy("The specifiedBy URL points to a human-readable specification. This field will only read a result for scalar types." url: String!) on SCALAR

"The `DateTime` scalar represents an ISO-8601 compliant date time type."
scalar DateTime @specifiedBy(url: "https:\/\/www.graphql-scalars.com\/date-time")

scalar UUID @specifiedBy(url: "https:\/\/tools.ietf.org\/html\/rfc4122")
